Allow tile metadata service to generate tilejson for scanned map with single raster

eliotjordan opened the pull request:
Works towards #5399 

![Screen Shot 2022-11-15 at 1 54 15 PM](https://user-images.githubusercontent.com/784196/202017737-6b54164e-e9b9-4bec-94d0-bf4ffe795b6d.png)

hackartisan commented on app/values/tile_path.rb:
```diff
@@ -48,13 +48,20 @@ def tileserver
     def mosaic?
-      return true unless resource.is_a?(RasterResource)
-      return true if resource.decorate.decorated_raster_resources.count.positive?
+      return true if resource.is_a?(RasterResource) && resource.decorate.raster_resources_count > 1
```
I think we'd like to prefer `Wayfinder.for(resource).raster_resources_count` over `resource.decorate.raster_resources_count`

hackartisan commented on app/services/tile_metadata_service.rb:
```diff
@@ -21,8 +21,8 @@ def path
   def mosaic?
-    return true unless resource.is_a?(RasterResource)
-    return true if resource.decorate.raster_resources_count.positive?
+    return true if resource.is_a?(RasterResource) && resource.decorate.raster_resources_count.positive?
```
Prefer using a wayfinder rather than a decorator here as well

hackartisan reviewed (changes requested):
It's not obvious to me why we're checking for positive or `> 1` in any given case. Maybe you could add comments to say why these are the checks we use?

eliotjordan commented:
@hackartisan I simplified the mosaic method, added some comments, and fixed some logic that was incorrect in our original implementation. A "mosaic" should have two or more raster files.

hackartisan commented on spec/services/tile_metadata_service_spec.rb:
```diff
@@ -63,14 +65,6 @@
       end
     end
-
-    context "when there aren't any files on the raster members" do
```
Why don't we want a test for this case anymore?

eliotjordan replied:
Good catch. We do, I just needed to move the `raise Error` elsewhere.

hackartisan reviewed (changes requested):
Thank you that is much more clear! I just have a question about this deleted test. Does it does something else now? Should we update it? Not sure why we want to delete it.
